src/plugins/smith_waterman/src/sw_cuda_cpp.cpp

```diff
@@ -176,34 +177,65 @@ QList<resType> calculateOnGPU(const char *seqLib, int seqLibLength, ScoreType *q
 
     //************************** declare arrays on device
 
-    char *g_seqLib;
-    ScoreType *g_queryProfile;
-    ScoreType *g_HdataMax;
-    ScoreType *g_HdataUp;
-    ScoreType *g_HdataRec;
+    char *g_seqLib = nullptr;
+    ScoreType *g_queryProfile = nullptr;
+    ScoreType *g_HdataMax = nullptr;
+    ScoreType *g_HdataUp = nullptr;
+    ScoreType *g_HdataRec = nullptr;
     ScoreType *g_HdataTmp;
-    ScoreType *g_FdataUp;
-    ScoreType *g_directionsUp;
-    ScoreType *g_directionsMax;
-    ScoreType *g_directionsRec;
+    ScoreType *g_FdataUp = nullptr;
+    ScoreType *g_directionsUp = nullptr;
+    ScoreType *g_directionsMax = nullptr;
+    ScoreType *g_directionsRec = nullptr;
     int *g_directionsMatrix = nullptr;
     int *g_backtraceBegins = nullptr;
 
     //************************** allocate global memory on device
 
-    cudaMalloc((void **)&g_seqLib, sizeL);
-    cudaMalloc((void **)&g_queryProfile, sizeP);
-    cudaMalloc((void **)&g_HdataMax, sizeQ);
-    cudaMalloc((void **)&g_HdataUp, sizeQ);
-    cudaMalloc((void **)&g_FdataUp, sizeQ);
-    cudaMalloc((void **)&g_directionsUp, sizeQ);
-    cudaMalloc((void **)&g_directionsMax, sizeQ);
-    cudaMalloc((void **)&g_HdataRec, sizeQ);
-    cudaMalloc((void **)&g_directionsRec, sizeQ);
+    QList<cudaError> errors = { cudaMalloc((void**)&g_seqLib, sizeL),
+                                cudaMalloc((void**)&g_queryProfile, sizeP),
+                                cudaMalloc((void**)&g_HdataMax, sizeQ),
+                                cudaMalloc((void**)&g_HdataUp, sizeQ),
+                                cudaMalloc((void**)&g_FdataUp, sizeQ),
+                                cudaMalloc((void**)&g_directionsUp, sizeQ),
+                                cudaMalloc((void**)&g_directionsMax, sizeQ),
+                                cudaMalloc((void**)&g_HdataRec, sizeQ),
+                                cudaMalloc((void**)&g_directionsRec, sizeQ) };
 
     if (U2::SmithWatermanSettings::MULTIPLE_ALIGNMENT == resultView) {
         cudaError errorMatrix = cudaMalloc(reinterpret_cast<void **>(&g_directionsMatrix), directionMatrixSize);
         cudaError errorBacktrace = cudaMalloc(reinterpret_cast<void **>(&g_backtraceBegins), backtraceBeginsSize);
+        errors << errorMatrix << errorBacktrace;
+    }
+
+    bool isError = false;
+    for (cudaError error : errors) {
```

Comment:
nit: find_if can fit the block 211...217 into 1 line

Reply:
I really want to use std, but I still don't understand when I can and when [I can't](https://github.com/ugeneunipro/ugene/pull/171#discussion_r556203899). Focus on the number of lines: if less than `for`, then use std?

Reply:
I see your point. The general suggestion is to select the most readable variant: maximum of your domain code and minimum of 'helper API boilerplate'.

You can compare the result in characters count or line count, but in general I think you can agree that this code (from your old PR):
```cpp
std::transform(inputPaths.begin(), inputPaths.end(), inputPaths.begin(), [](const QString &path) {
    return QFileInfo(path).absoluteFilePath();
});
```

Looks less readable than the suggested one:
```cpp
for (QString& path: inputPaths) {
    path = QFileInfo(path).absoluteFilePath();
}
```

 C++ has very verbose lambda syntax. In other languages I would prefer using lambdas, not an old-style loop for the same code.

Example (JavaScript):
```cpp
inputPath = inputPath.map(path => new QFileInfo(path).absoluteFilePath());
```

Or even shorter (Kotlin):
```cpp
inputPath = inputPath.map{ QFileInfo(it).absoluteFilePath() }
```

Below is the example what I call 'helper API boilerplate'. The same code but with our logic removed.

```cpp
// Variant 1:
std::transform(.begin(), .end(), .begin(), [](const QString &path) {
    return path;
});

// Variant 2:
for (QString& path: ) {
    path = path;
}

// JavaScript:
= .map(path => new path);

// Kotlin:
= .map{ it }
````